application/src/main/java/org/thingsboard/server/service/telemetry/DefaultTelemetrySubscriptionService.java

```diff
@@ -176,11 +182,37 @@ public void saveAttributes(AttributesSaveRequest request) {
     @Override
     public void saveAttributesInternal(AttributesSaveRequest request) {
         log.trace("Executing saveInternal [{}]", request);
-        ListenableFuture<List<Long>> saveFuture = attrService.save(request.getTenantId(), request.getEntityId(), request.getScope(), request.getEntries());
-        DonAsynchron.withCallback(saveFuture, result -> {
-            calculatedFieldQueueService.pushRequestToQueue(request, result, request.getCallback());
-        }, safeCallback(request.getCallback()), tsCallBackExecutor);
-        addWsCallback(saveFuture, success -> onAttributesUpdate(request.getTenantId(), request.getEntityId(), request.getScope().name(), request.getEntries(), request.isNotifyDevice()));
+        TenantId tenantId = request.getTenantId();
+        EntityId entityId = request.getEntityId();
+        AttributesSaveRequest.Strategy strategy = request.getStrategy();
+        ListenableFuture<List<Long>> resultFuture;
+
+        if (strategy.saveAttributes()) {
+            resultFuture = attrService.save(tenantId, entityId, request.getScope(), request.getEntries());
+        } else {
+            resultFuture = Futures.immediateFuture(Collections.emptyList());
+        }
+
+        addMainCallback(resultFuture, result -> {
+            if (strategy.processCalculatedFields()) {
+                calculatedFieldQueueService.pushRequestToQueue(request, result, request.getCallback());
+            } else {
+                request.getCallback().onSuccess(null);
+            }
+        }, t -> request.getCallback().onFailure(t));
+
+        if (strategy.saveAttributes()
+                && entityId.getEntityType() == EntityType.DEVICE
+                && TbAttributeSubscriptionScope.SHARED_SCOPE.name().equalsIgnoreCase(request.getScope().name())
+                && request.isNotifyDevice()) {
+            addMainCallback(resultFuture, success -> clusterService.pushMsgToCore(
+                    DeviceAttributesEventNotificationMsg.onUpdate(tenantId, new DeviceId(entityId.getId()), DataConstants.SHARED_SCOPE, request.getEntries()), null
+            ));
```

Comment:
What about attribute deletion? Maybe we should also move the logic from DefaultSubscriptionManagerService#onAttributesDelete to this service?